bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/handlers/FileOpenListener.java

```diff
@@ -34,6 +35,10 @@ public void partActivated(IWorkbenchPartReference partRef)
 		if (part instanceof IEditorPart ieditorpart
 				&& ieditorpart.getEditorInput() instanceof FileEditorInput fileInput)
 		{
+			
+			if (ieditorpart instanceof ESPIDFManagerEditor)
+				return;
```

Comment:
We can make it more robust because, theoretically, the same exception could occur in another editor. What if we get the content description from the file (fileInput.getFile()) and then compare the content type ID with the following condition: id.startsWith("org.eclipse.cdt.core.c") && (id.endsWith("Source") || id.endsWith("Header"))?

Reply:
That is actually a very nice idea let me try to update this :)